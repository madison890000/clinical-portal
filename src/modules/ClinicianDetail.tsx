import { useState } from 'react';
import { Tab, Tabs, Typography } from '@mui/material';


const TabPanel = ({ children, value, index }: any) => {
    return (
        <>
            {value === index && children}
        </>
    )
}

const ClinicianDetail = () => {
    const [activeTab, setActiveTab] = useState<number>();
    const handleChange = (event: React.SyntheticEvent, newValue: number) => {
        setActiveTab(newValue);
    };
    return (
        <>
            <div style={{
                display: 'flex',
                alignItems: 'center',
            }}>
                <div style={{
                    margin: '40px',
                    height: 200,
                }}>
                    <Typography align="left">Clinical Portal</Typography>
                </div>
                <div>info</div>
            </div>

            <Tabs value={activeTab} onChange={handleChange}>
                <Tab label="item1" />
                <Tab label="item2" />
            </Tabs>
            <TabPanel value={activeTab} index={0}>
                Item One
            </TabPanel>
            <TabPanel value={activeTab} index={1}>
                Item Two
            </TabPanel>
        </>
    )
}

export default ClinicianDetail
