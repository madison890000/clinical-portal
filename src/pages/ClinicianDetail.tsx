import { useCallback, useContext, useState } from 'react';
import { Divider, Skeleton, Tab, Tabs, Typography } from '@mui/material';
import PatientDetail from '../modules/PatientDetail';
import ClinicianInfo from '../modules/ClinicianInfo';
import { AppContext } from '../contexts/AppContext';


const TabPanel = ({ children, value, index }: any) => {
    return (
        <div style={{ display: value === index ? 'block' : 'none' }}>
            {children}
        </div>
    )
}

const ClinicianDetail = () => {
    const { clinician, patients } = useContext(AppContext);
    const [activeTab, setActiveTab] = useState<number>(0);
    const handleChange = useCallback((event: React.SyntheticEvent, newValue: number) => {
        setActiveTab(newValue);
    }, [setActiveTab]);
    return (
        <>
            <div style={{
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'space-between',
                margin: 20
            }}>
                <div style={{
                    margin: '40px',
                    flex: 'auto',
                    textAlign: 'center'
                }}>
                    <Typography variant="h4" align="center">Clinical Portal</Typography>
                </div>
                {!clinician && <Skeleton variant="rectangular" width={300} height={118} />}
                {clinician && <ClinicianInfo {...clinician} />}
            </div>

            <Divider />
            <Tabs value={activeTab} onChange={handleChange}>
                {
                    patients?.map(patient => (
                        <Tab style={{ minWidth: 500 }} label={`${patient.name} (${patient.id})`}
                             key={patient.id} value={patient.id} />
                    ))
                }
            </Tabs>
            {
                patients?.map(patient => (
                    <TabPanel value={activeTab} index={patient?.id}>
                        <PatientDetail id={patient?.id} />
                    </TabPanel>
                ))
            }
        </>
    )
}

export default ClinicianDetail
