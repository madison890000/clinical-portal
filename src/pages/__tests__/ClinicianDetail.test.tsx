import React from 'react';
import ClinicianDetail from '../ClinicianDetail';
import { render, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

test('renders ClinicianDetail Page', async () => {
    const { container } = render(
        <BrowserRouter>
            <ClinicianDetail />
        </BrowserRouter>
    );
    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});
