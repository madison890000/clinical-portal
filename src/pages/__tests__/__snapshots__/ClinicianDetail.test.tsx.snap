// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ClinicianDetail Page 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="portalTitleContainer"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
      >
        Clinical Portal
      </h4>
    </div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
      style="width: 300px; height: 118px;"
    />
  </div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <div
    class="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      />
      <span
        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
